thonny/plugins/autocomplete.py

```diff
@@ -70,7 +72,21 @@ def _handle_backend_response(self, msg):
         else:
             self._present_completions(msg.completions)
 
-    def _present_completions(self, completions):
+    def _present_completions(self, completions_):
+        # Check if autocompeted name starts with an underscore,
+        # if it doesn't - don't show names starting with '_'
+        source = self.text.get("1.0", "end-1c")
```

Comment:
It looks like you are assuming, the user is appending to the source, but in reality, the insertion point can be in the middle of the source.

I suggest you ask for text from the start of the current line up to the insertion cursor: `self.text.get("insert linestart", "insert")`